### PR TITLE
Implement game objects removal, move out charge logic from weapons

### PR DESCRIPTION
**Pull request by pkulev**
**Closes issue(s):** [ #59 ]
**Related to issue(s):** [ #52 ]

**Description of the changes:**

This PR makes two major changes:
* Split Weapon into Weapon and Charge (as separate renderable entities)
* Implement game objects removal

Both of them lead us to adding GAMEPLAY! Pew-pew and homebrew gameovers can be implemented as well as exposions. This is really great good news.

  * What's new:
    * weapon and charge are separate entities
    * remove objects via State's remove method (now it's only InGame state, but will be moved soon)
    * current Application accessible via application.get_current()
  * Fixes:
    * fix logging to stdout
    * fix formatting, docstrings

  * Refactoring:
    * refactor weapon

### CODE REVIEW

File: xoinvader/render.py [hunk 1]
```diff
@@ -52,18 +66,19 @@ def render_objects(objects, screen):
         for data in data_gen:
             for gpos in gpos_list:
                 lpos, image, style = data
-                cpos = gpos + lpos
-
-                if (cpos.x >= border.x or cpos.y >= border.y) \
-                   or (cpos.x <= 0 or cpos.y <= 0):
+                cpos = (gpos + lpos)[int]
```
Comment (taptap) [inline xoinvader/render.py]: what

File: xoinvader/render.py [hunk 2]
```diff
@@ -15,12 +15,15 @@ class Renderable(object):
...
 
+    * compound (bool): if object consists of other renderables
+
     * render_priority (int): priority for renderer, greater -> rendered later
 
     * draw_on_border (bool): allow or not drawing on border
     if not allowed - renderer will send remove_obsolete signal to object
     """
 
+    compound = False
```
Comment (taptap) [inline xoinvader/render.py]: Why add flag? Can't `get_renderable_objects` just return empty list?

File: xoinvader/config/path.json [hunk 3]
```diff
@@ -1,25 +1,26 @@
 {
-    "config" : {
-        "ships" : "/config/ships.json",
-        "weapons" : "/config/weapons.json"
+    "config": {
+        "charges": "/config/charges.json",
```
Comment (pkulev) [inline xoinvader/config/path.json]: fixed

File: xoinvader/render.py (hunk 1 above)
Comment (pkulev) [inline xoinvader/render.py]: 3877ad304b68d981830123d92164369e870e5d9c

Comment (pkulev) [inline xoinvader/render.py]: 3877ad304b68d981830123d92164369e870e5d9c

File: xoinvader/render.py (hunk 2 above)
Comment (pkulev) [inline xoinvader/render.py]: Such change leads to 2 extra function calls and logging empty list for Background and GUI elements in one moment and for charges every shot. This is not so bad but I don't see any needs to do so (now).

File: xoinvader/render.py (hunk 1 above)
Comment (taptap) [inline xoinvader/render.py]: Ok!

File: xoinvader/render.py (hunk 2 above)
Comment (taptap) [inline xoinvader/render.py]: Ok.

File: xoinvader/config/path.json (hunk 3 above)
Comment (taptap) [inline xoinvader/config/path.json]: Ok.

**Pull request merged**